corehq/apps/locations/expressions.py

```diff
@@ -0,0 +1,35 @@
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.userreports.expressions.factory import ExpressionFactory
+from corehq.apps.userreports.specs import TypeProperty
+from corehq.util.quickcache import quickcache
+from dimagi.ext.jsonobject import JsonObject, DictProperty
+
+
+class LocationTypeSpec(JsonObject):
+    type = TypeProperty('location_type_name')
+    location_id_expression = DictProperty(required=True)
+
+    def __call__(self, item, context=None):
+        location_id_expression = ExpressionFactory.from_spec(self.location_id_expression)
+        doc_id = location_id_expression(item, context)
+        if not doc_id:
+            return None
+
+        assert context.root_doc['domain']
+        return self.get_location_type(doc_id, context, context.root_doc['domain'])
+
+    @quickcache(['location_id', 'domain'], timeout=600)
+    def get_location_type(self, location_id, context, domain):
```

Comment:
can you make this a `staticmethod` or just a function? no need for it to be part of the class